__tests__/ship.test.js

```diff
@@ -0,0 +1,100 @@
+/* globals describe it expect */
+const Ship = require('../src/ship');
+const Port = require('../src/port');
+const Itinerary = require('../src/itinerary');
+
+describe('Ship constructor', () => {
+  describe('Ships with ports and an itinerary', () => {
+    let ship;
+    let dover;
+    let calais;
+    let itinerary;
+    // let port;
+
+    beforeEach(() => {
+      dover = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: 'Dover',
+        ships: []
+      };
+
+      calais = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: 'Calais',
+        ships: []
+      };
+      port = new Port('Dover'); 
+      // dover = new Port('Dover'); replaced with spies above.
+      // calais = new Port('Calais'); replaced with spies above.
+      itinerary = new Itinerary ([dover, calais]);
+      ship = new Ship(itinerary);
+    });
+
+    it('Ship can be instantiated', () => {
+      const port = new Port('Dover');
+      const itinerary = new Itinerary([port]);
+      const ship = new Ship(itinerary);
+
+      expect(ship).toBeInstanceOf(Object);
+    });
+ 
+    it('Ship has a starting port', () => {
+      // const ship = new Ship('Dover');
+      // const dover = new Port('Dover');
+      const itinerary = new Itinerary([port]);
+      const ship = new Ship(itinerary); //cannot delete this line, test fails - why??
+
+      expect(ship.currentPort).toBe(port);
+    });
```

Comment:
Okay, so I think there might be a bit of confusion about how the test is working here. As above, you are creating variables for things that already exist with itinerary and ship already being defined above. This means you can delete all the content of this function before the assertion as it already has everything it needs

The test will now fail though, so why is this? Looking at the beforeEach you can see that you should be expecting the current port to be dover, but in the test you are asserting for `port`. So we can fix this with:
```
    it('Ship has a starting port', () => {
      expect(ship.currentPort).toBe(dover);
    });
```